confidant/models/credential.py

```diff
@@ -142,6 +161,41 @@ def _get_decrypted_credential_pairs(self):
         _credential_pairs = json.loads(_credential_pairs)
         return _credential_pairs
 
+    @property
+    def next_rotation_date(self):
+        """
+        Return when a credential needs to be rotated for security purposes.
+        """
+        # Some credentials never need to be rotated
+        if self.exempt_from_rotation:
+            return None
+
+        # If a credential has never been rotated or been decrypted,
+        # immediately rotate
+        if self.last_rotation_date is None:
+            return datetime.utcnow()
+
+        if (self.last_decrypted_date and
+                self.last_decrypted_date > self.last_rotation_date):
+            return datetime.utcnow()
+
+        days = settings.MAXIMUM_ROTATION_DAYS
+        for tag in self.tags:
+            rotation_days = settings.ROTATION_DAYS_CONFIG.get(tag)
+            if rotation_days is None:
+                continue
+            if days is None or rotation_days < days:
+                days = rotation_days
+        return self.last_rotation_date + timedelta(days=days)
```

Comment:
days can be None here still. I'd imagine we want the behavior for this to not flag a credential for rotation if settings.MAXIMUM_ROTATION_DAYS isn't set, or if no tags are applied?

Reply:
Yea I'm aware of this - I wanted to specify this in the readme.  I could make a check for that if you feel strongly

```
if self.exempt_from_rotation or settings.MAXIMUM_ROTATION_DAYS is None:
    return None
```

Reply:
Code should always be defensive. If the default for `settings.MAXIMUM_ROTATION_DAYS` is None, this is going to throw an exception:

```
>>> from datetime import timedelta
>>> timedelta(days=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported type for timedelta days component: NoneType
```

Reply:
Oh, it's using int_env, so I guess the default is 0. This would mean all credentials would always show as needing rotation, but I guess if people aren't using the feature, it's not an issue.

Reply:
Yea it shouldn't cause problems if people aren't using the feature and the README will say very explicitly that MAXIMUM_ROTATION_DAYS must be set